### DISCUSSION TITLE
Generating Web Assembly for opt tool

### DISCUSSION THREAD

**Discussion by talhatahir-10xe** [MLIR-China/mlir-playground#81] [Q&A]
Hi! How did you generate WebAssembly for the MLIR opt tool? While it's relatively straightforward to generate .wasm files for simple C/C++ programs using Emscripten, projects like MLIR often have many files and dependencies. I tried to compile LLVM from Emscripten using emmake with the command:
`emmake cmake -G "Unix Makefiles" -DLLVM_ENABLE_PROJECTS="llvm" -DCMAKE_BUILD_TYPE=Release ../llvm`
However, I encountered continuous errors. Can you provide guidance on how you compiled mlir-opt using Emscripten?

**Comment by zyx-billy** [comment]
Hi, yeah it was certainly not easy getting it a large project to work 😂 but we have a docker setup now [here](https://github.com/MLIR-China/mlir-playground/blob/main/cpp/docker/Dockerfile) that will build an image that contains LLVM & MLIR libraries & binaries built for WebAssembly. You just need to copy the files you want out of it.
If you're interested in reproducing it locally or trying a different llvm version, you can try following the steps manually in the `setup.sh` & `install.sh` scripts used by the dockerfile. Hopefully that helps!
